docker/importer/importer.py

```diff
@@ -704,35 +704,44 @@ def _process_updates_rest(self, source_repo: osv.SourceRepository):
     """Process updates from REST API."""
     logging.info('Begin processing REST: %s', source_repo.name)
 
-    ignore_last_import_time = (
-        source_repo.ignore_last_import_time or not source_repo.last_update_date)
-    if ignore_last_import_time:
+    last_update_date = source_repo.last_update_date or datetime.datetime.min
+    if source_repo.ignore_last_import_time:
+      last_update_date = datetime.datetime.min
       source_repo.ignore_last_import_time = False
       source_repo.put()
-    import_time_now = utcnow()
+
     request = requests.head(source_repo.rest_api_url, timeout=_TIMEOUT_SECONDS)
     if request.status_code != 200:
       logging.error('Failed to fetch REST API: %s', request.status_code)
       return
 
-    if 'Last-Modified' in request.headers:
-      last_modified = datetime.datetime.strptime(
-          request.headers['Last-Modified'], _HTTP_LAST_MODIFIED_FORMAT)
-      # Check whether endpoint has been modified since last update
-      if not ignore_last_import_time and (last_modified
-                                          < source_repo.last_update_date):
-        logging.info('No changes since last update.')
-        return
+    request_last_modified = None
+    if last_modified := request.headers.get('Last-Modified'):
+      try:
+        request_last_modified = datetime.datetime.strptime(
+            last_modified, _HTTP_LAST_MODIFIED_FORMAT)
+        # Check whether endpoint has been modified since last update
+        if request_last_modified <= last_update_date:
+          logging.info('No changes since last update.')
+          return
+      except ValueError:
+        logging.error('Invalid Last-Modified header: "%s"', last_modified)
 
     request = requests.get(source_repo.rest_api_url, timeout=_TIMEOUT_SECONDS)
```

Comment:
I've tried to do this in 085e161 using [requests / urllib3](https://requests.readthedocs.io/en/latest/user/advanced/#example-automatic-retries), to avoid having to add a new dependency.